Source/WebCore/Modules/plugins/YouTubePluginReplacement.cpp

```diff
@@ -76,16 +76,20 @@ RenderPtr<RenderElement> YouTubePluginReplacement::createElementRenderer(HTMLPlu
     if (!m_embedShadowElement)
         return nullptr;
     
-    return m_embedShadowElement->createElementRenderer(WTFMove(style), insertionPosition);
+    return Ref { *m_embedShadowElement }->createElementRenderer(WTFMove(style), insertionPosition);
 }
 
 void YouTubePluginReplacement::installReplacement(ShadowRoot& root)
 {
-    m_embedShadowElement = YouTubeEmbedShadowElement::create(m_parentElement->document());
+    Ref document = m_parentElement->document();
 
-    root.appendChild(*m_embedShadowElement);
+    m_embedShadowElement = YouTubeEmbedShadowElement::create(document);
```

Comment:
I often go for:
```
Ref embedShadowElement = YouTubeEmbedShadowElement::create(document);
m_embedShadowElement = embedShadowElement.copyRef();
```

It is nice before it makes it clear we have a `Ref` and we don't have to do `*m_embedShadowElement` to convert a RefPtr to a Ref.

Reply:
>     // The following function is deprecated.

Maybe it shouldn't be deprecated?